webapp/channels/src/components/channel_header_mobile/show_search_button/show_search_button.tsx

```diff
@@ -15,24 +15,26 @@ type Props = {
     actions: Actions;
 }
 
-export default class ShowSearchButton extends React.PureComponent<Props> {
-    handleClick = () => {
-        this.props.actions.openRHSSearch();
-    };
-
-    render() {
-        return (
-            <button
-                type='button'
-                className='navbar-toggle navbar-right__icon navbar-search pull-right'
-                onClick={this.handleClick}
-                aria-label={localizeMessage('accessibility.button.Search', 'Search')}
-            >
-                <SearchIcon
-                    className='icon icon__search'
-                    aria-hidden='true'
-                />
-            </button>
-        );
-    }
-}
+const ShowSearchButton = ({actions}: Props) => {
+    const {openRHSSearch} = actions;
+
+    const handleClick = useCallback(() => {
+        openRHSSearch();
+    }, [openRHSSearch]);
+
+    return (
+        <button
+            type='button'
+            className='navbar-toggle navbar-right__icon navbar-search pull-right'
+            onClick={handleClick}
+            aria-label={localizeMessage('accessibility.button.Search', 'Search')}
+        >
+            <SearchIcon
+                className='icon icon__search'
+                aria-hidden='true'
+            />
+        </button>
+    );
+};
+
+export default ShowSearchButton;
```

Comment:
Sorry, I'm also missing here the `React.memo` call, because the original one was a `PureComponent`:
```suggestion
export default React.memo(ShowSearchButton);
```

Reply:
Done the changes